core/include/gnuradio-4.0/Block.hpp

```diff
@@ -1292,6 +1137,223 @@ class Block : public lifecycle::StateMachine<Derived>, //
+        return std::tuple{hasTag, nextTag, nextEosTag, asyncEoS};
+    }
+
+    /***
+     * skip leftover stride
+     * @param available number of samples that can be consumed from each sync port
+     * @return inputSamples to skip before the chunk
+     */
+    std::size_t
+    inputSamplesToSkipBeforeNextChunk(std::size_t available) {
+        std::size_t inputSamplesToSkipBefore = 0UZ;
+        if constexpr (StrideControl::kEnabled) {                    // check if stride was removed at compile time
+            if (stride.value != 0 && stride.value != denominator) { // check if stride is disabled at runtime
+                if (strideCounter != 0) {                           // drop samples
+                    if (strideCounter >= static_cast<gr::Size_t>(available)) {
+                        inputSamplesToSkipBefore = available;
+                        strideCounter -= static_cast<gr::Size_t>(available);
+                    } else {
+                        inputSamplesToSkipBefore = strideCounter;
+                        strideCounter            = 0;
+                    }
+                }
+            }
+        }
+        return inputSamplesToSkipBefore;
+    }
+
+    /***
+     * skip leftover stride
+     * @return inputSamples to skip before the chunk
+     */
+    std::size_t
+    inputSamplesToSkipAfterThisChunk(std::size_t remaining) {
+        // skip stride at the end
+        std::size_t inputSamplesToSkipAfter = 0UZ;
+        if constexpr (StrideControl::kEnabled) {
+            if (stride.value != 0 && stride.value != denominator) {
+                if (strideCounter == 0 && remaining > 0) {
+                    inputSamplesToSkipAfter = std::min(static_cast<std::size_t>(stride.value), remaining);
+                    strideCounter           = stride.value - static_cast<gr::Size_t>(inputSamplesToSkipAfter);
+                }
+            }
+        }
+        return inputSamplesToSkipAfter;
+    }
+
+    auto
+    decimate(std::size_t minSyncIn, std::size_t maxSyncIn, std::size_t minSyncOut, std::size_t maxSyncOut) {
+        if constexpr (Resampling::kEnabled) {
+            if (denominator != 1UL || numerator != 1UL) {
+                std::size_t nResamplingChunks;
+                if constexpr (StrideControl::kEnabled) { // with stride, we cannot process more than one chunk
+                    if (stride.value != 0 && stride.value != denominator) {
+                        nResamplingChunks = denominator <= maxSyncIn && numerator <= maxSyncOut ? 1 : 0;
+                    } else {
+                        nResamplingChunks = std::min(maxSyncIn / denominator, maxSyncOut / numerator);
+                    }
+                } else {
+                    nResamplingChunks = std::min(maxSyncIn / denominator, maxSyncOut / numerator);
+                }
+                if (nResamplingChunks * denominator < minSyncIn || nResamplingChunks * numerator < minSyncOut) {
```

Comment:
We might need to incorporate additional conditions to handle additional ill-formed parameters related to `numerator` and `denominator`. For instance, consider the scenario where `minSyncIn = 2`, `maxSyncIn = 3`, and `denominator = 2`. In this case, `nResamplingChunks = 1`, leading to a situation where it consistently returns {0, 0}, potentially resulting in an infinite loop.